pyccel/codegen/printing/ccode.py

```diff
@@ -391,10 +391,7 @@ def _print_NumpyUfuncBase(self, expr):
         func_name = numpy_ufunc_to_c[type_name]
         args = []
         for arg in expr.args:
-            if arg.dtype is not NativeReal():
-                args.append('(' + self._print_PythonFloat(arg) + ')')
-            else:
-                args.append(self._print(arg))
+            args.append(self._print(arg))
```

Comment:
I don't know why you removed these conditions. this will raise a warning if you pass an integer to one of these functions because they accept only float as an argument. btw you can't see this warning in pytest epyccel tests, to see this warning you should test this case by pyccel command

Reply:
I have modified the compiler call. These warnings should be visible when using pytest and epyccel now. They are not always raised on my computer though, it may be worth using the '-Wall' flag. But if we do that it is possible that there will be lots of warnings that appear, so we should probably save that for a new PR.

Reply:
The additional warnings that are appearing currently are due to array tests being enabled without arrays being implemented in C. Apparently the cwrapper doesn't raise an error if we try to use the arrays so we just generate code that is wrong

Reply:
i will change that to be casted explicitly

Reply:
what do you think if we write these warnings in a log file?
instead of show them exactly we can show only the number of warnings and the path to warnings log file.

Reply:
I don't think that's a good idea. It's really easy to ignore a log file and warnings are often indicative of a more important problem. In theory pyccel should generate warning free code so the only warnings should be due to the user's poorly written code.

As an example of why it is good to keep the warnings: now they are enabled we can see warnings about a line `a == b != 0` which is a translation of `a is bool(b)`. For bools it should work, but if it had been `a == bool(b)` with `a` and `b` being integers then this could lead to incorrect results I think.